src/components/Modal/Modal.tsx

```diff
@@ -76,7 +77,7 @@ const Modal: React.FC<Props> = ({ open, onClose, children, AnimationComponent =
 
   return ReactDOM.createPortal(
     <Fade open={open} duration={300} onCloseAnimationEnd={() => setVisible(false)}>
-      <div className={styles.modal} onKeyDown={keyDownEventHandler} ref={modalRef}>
+      <div className={styles.modal} onKeyDown={keyDownEventHandler} ref={modalRef} role={role}>
```

Comment:
Can we not just always set the role to "dialog" so we don't need a prop? I think every modal is a dialog (just [not every dialog is a model](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role), strictly speaking). 
But if you do need to use a prop for this, then I have a few suggestions/considerations:
- I think we should set the role to the Fade component, cause this creates a wrapping div
- For typical html props such as role, I think instead of introducing a new prop, we should use the standard typing: `type Props = HTMLDialogElement & {`, etc

Reply:
We should add the `aria-modal="true"` attribute and can set the default `role` prop to `alert`. But we should keep the option open to be able to change it to `alertdialog` and `alert`.

The role attribute is not expected on the wrapping div, it should be added to the dialog only. See https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/

We probably should not use `HTMLDialogElement` because we're not actually using an HTML dialog element. React does have `React.AriaAttributes` and `React.AriaRole` however. We can cherry-pick some properties from there instead of adding them all.

```ts
type Props = Pick<AriaAttributes, 'aria-labelledby' | 'aria-label'> & {
  open: boolean;
  onClose: () => void;
  size?: 'small' | 'large';
  children: React.ReactNode;
  role: React.AriaRole;
};
```

But this is primarily useful when spreading props to the element directly:

```tsx
const Dialog: React.FC<Props> = ({ open, onClose, size = 'small', children, ...rest }: Props) => {
  return (
    <Modal open={open} onClose={onClose} AnimationComponent={Slide} {...rest}>
```

We can use this opportunity to make the `aria-labelledby` prop required. So for now, I would add it specifically:

```tsx
type Props = {
  open: boolean;
  onClose: () => void;
  size?: 'small' | 'large';
  children: React.ReactNode;
  role: React.AriaRole;
  'aria-labelledby': string; // non-optional 
};
```

Reply:
That's a great complete reply!
I only doubt about the default `alert` role, shouldn't we default to `dialog`?

Reply:
Ah yes, I meant setting the default `role` to `dialog` 😄

Reply:
After discussion with the team, we've not made the aria-labelledby prop required because:
1. In the `AccountModal` component we have the `<Dialog>` component, but this renders many components for which we need to set the `id` attribute, this will take up more time and can even cause duplicate id's in some cases. 
2. It is not a requirement and therefore not seen as important enough for this stage